### DISCUSSION TITLE
https://duckduckgo.com Exception: Error in request: error sending request for url (https://duckduckgo.com/): operation timed out

### DISCUSSION THREAD

**Discussion by windkwbs** [open-webui/open-webui#3118] [General]
![微信截图_20240613094753](https://github.com/open-webui/open-webui/assets/129468439/a04e69f2-befd-4189-8554-2ca87d871539)
https://duckduckgo.com Exception: Error in request: error sending request for url (https://duckduckgo.com/): operation timed out

**Comment by lodern** [comment]
Try to use **docker exec -it image_id /bin/bash**
If the internet connection is fine, not in GFW, try to uninstall duckduckgo_search package of python
 **pip uninstall duckduckgo_search**
and then reinstall it
 **pip uninstall duckduckgo_search**
If you're in GFW of CN, try to set proxy before reinstall it as below:
**export https_proxy=http://your_proxy_ip:port_number
export https_proxy=http://your_proxy_ip:port_number
export no_proxy="127.0.0.1,your.local.domain"**
It should be solved. 
Addison

**Reply by syawilbur110** [reply]
i m in GFW of CN, and using open webui, how to set up ?

**Reply by lodern** [reply]
Entering bash shell of openwebui and trying to
add http_proxy and https_proxy in ~/.bashes then only restart  docker image of it.
If it does work, save docker image to be effected  permanently.

**Reply by code-monad** [reply]
I am running into the same issue.
The env var i have already set. Also i am pretty sure the `Trust Proxy Environment` option has been checked.

still, duckduckgo will always went into a operation timed out error.

But i can visit every other stuff via this proxy, including the api endpoint, or other curl requests to google etc.

**Comment by lodern** [comment]
Try to config proxy’s settings in both ~/.bashed and start.sh on docker image of openwebui, upgrade duckduckgo module with pip install —upgrade duckduckgo-search, and finally restart openwebui with docker restart command.
